src/test/java/org/jabref/logic/importer/DatabaseFileLookupTest.java

```diff
@@ -1,47 +1,145 @@
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 class DatabaseFileLookupTest {
 
+    @TempDir
+    Path tempDir;
     private BibDatabase database;
+    private String txtFileDirPath;
     private Collection<BibEntry> entries;
 
     private BibEntry entry1;
     private BibEntry entry2;
+    private Path txtFileDir;
+    private MetaData metaData;
+    private BibDatabaseContext databaseContext;
+    private FilePreferences filePreferences;
+    private DatabaseFileLookup fileLookup;
 
     @BeforeEach
     void setUp() throws Exception {
-        ParserResult result = new BibtexImporter(mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS), new DummyFileUpdateMonitor())
+        ParserResult result = new BibtexImporter(mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS),
+                new DummyFileUpdateMonitor())
                 .importDatabase(ImportDataTest.UNLINKED_FILES_TEST_BIB);
         database = result.getDatabase();
         entries = database.getEntries();
 
-        entry1 = database.getEntryByCitationKey("entry1").get();
-        entry2 = database.getEntryByCitationKey("entry2").get();
+        txtFileDir = tempDir.resolve("x.txt");
+        txtFileDirPath = txtFileDir.toAbsolutePath().toString();
+        Files.write(txtFileDir, Collections.singleton("x.txt file contents for test"));
+
+        entry1 = database.getEntryByCitationKey("entry1")
+                .orElseThrow(() -> new Exception("Entry with citation key 'entry1' not found"));
+
+        entry2 = database.getEntryByCitationKey("entry2")
+                .orElseThrow(() -> new Exception("Entry with citation key 'entry2' not found"));
+
+        BibEntry entry3 = new BibEntry().withField(StandardField.FILE, txtFileDirPath);
+        BibEntry entry4 = new BibEntry().withField(StandardField.FILE, "");
+
+        List<BibEntry> entries = List.of(entry1, entry2, entry3, entry4);
+        database = new BibDatabase(entries);
+
+        filePreferences = mock(FilePreferences.class);
+        when(filePreferences.getMainFileDirectory()).thenReturn(Optional.of(txtFileDir.toAbsolutePath()));
+        when(filePreferences.getUserAndHost()).thenReturn("User124");
+
+        metaData = new MetaData();
+        metaData.setDefaultFileDirectory(txtFileDirPath);
+        metaData.setUserFileDirectory("User124", txtFileDirPath);
+
+        databaseContext = new BibDatabaseContext(database, metaData, txtFileDir.toAbsolutePath());
+        databaseContext.setDatabasePath(txtFileDir);
+
+        fileLookup = new DatabaseFileLookup(databaseContext, filePreferences);
     }
 
     /**
      * Tests the prerequisites of this test-class itself.
      */
     @Test
     void prerequisitesFulfilled() {
-        assertEquals(2, database.getEntryCount());
+        assertEquals(4, database.getEntryCount());
         assertEquals(2, entries.size());
         assertNotNull(entry1);
         assertNotNull(entry2);
```

Comment:
A test case should focus on **one** thing. Not testing that constructors of data classes work. Remove all three assertions.